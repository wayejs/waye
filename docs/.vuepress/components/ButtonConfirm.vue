<template>
  <div>
    <wy-button-confirm>删除</wy-button-confirm>
  </div>
</template>

<script>
export default {
  name: 'ButtonConfirm',
  data () {
    return {
    }
  }
}
</script>
