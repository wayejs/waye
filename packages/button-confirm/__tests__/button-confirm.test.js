import { shallowMount } from '@vue/test-utils'
import sinon from 'sinon'
import ButtonConfirm from '../button-confirm'
const clickHandler = sinon.stub()
describe('ButtonConfirm', () => {
  it('init ButtonConfirm', async () => {
    const wrapper = shallowMount(ButtonConfirm, {
      propsData: {
        type: 'primary',
        clickHandler
      },
      slots: {
        default: 'delete'
      },
      attachToDocument: true
    })

    wrapper.find({ name: 'ElButton' }).trigger('click')
    // document.body.click()

    // expect(clickHandler.called).toBe(true)
    // expect(wrapper).toMatchSnapshot()
    // expect(document.querySelector('.el-message-box__wrapper').length).toBe(1)
    wrapper.destroy()
  })
})
