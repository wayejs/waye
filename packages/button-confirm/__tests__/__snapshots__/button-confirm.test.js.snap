// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonConfirm init ButtonConfirm 1`] = `
<div class="wy-button-confirm" clickhandler="stub">
  <el-button-stub type="primary" icon="" nativetype="button">delete</el-button-stub>
</div>
`;
